smoke-tests/build.gradle

```diff
@@ -14,6 +14,10 @@ configurations {
   }
 }
 
+compileTestJava {
+  options.release.set(17)
```

Comment:
[For Reviwer] Note that it is not strictly required that all tests are Java 8 compatible.
It is possible to simply override the minimum testing java version on a per-project basis like here.
In this case the smoke-tests won't be executed for Java versions older than 17

Reply:
[question] how does those smoke tests get skipped and reported when executing tests on java8 ?
- are they absent, failed (but ignored) or marked as skipped in the test report ?
- do we have to do anything specific to make them ignored (or executed) depending on the JVM used to run them ? Is there any kind of opt in/out or is it transparently managed by the toolchain in gradle ?

Reply:
The corresponding gradle `test` task will be simply disabled, so the tests won't be loaded by the target JVM and therefore don't appear in any state in the test reports.

See this line for the logic:
https://github.com/elastic/elastic-otel-java/pull/187/files#diff-8c76762546ca458066ae00e3880c724d446b6b0cad63cf585c71a9d27b91a0fbR19

If you write your tests Java 8 compilable, but only want them to run on specific JVMs you can use the JUnit annotations on your tests instead. This case is just for where we want to use libraries or Language features in tests which cannot be compiled to Java 8. In this case, the smoke tests use Text-blocks for readability and it doesn't make sense to run the smoketests on older versions, because the actual testing happens inside containers anyway.

If we wanted to be super clean, we could introduce a separate task + workflow job for the smoke tests, but this is not really worth it now IMO.